aws/ec2metadata/api_test.go

```diff
@@ -288,6 +292,19 @@ func TestGetMetadata(t *testing.T) {
 			expectedError:               "404",
 			expectedOperationsAttempted: []string{"GetToken", "GetMetadata", "GetMetadata"},
 		},
+		"invalid token request case": {
+			NewServer: func(t *testing.T) *httptest.Server {
+				testType := InvalidTokenRequestTestType
+				Ts := &testServer{
+					t:      t,
+					tokens: []string{"firstToken", "secondToken", "thirdToken"},
+					data:   "IMDSProfileForGoSDK",
+				}
+				return newTestServer(t, testType, Ts)
+			},
+			expectedError:               "401",
+			expectedOperationsAttempted: []string{"GetToken", "GetMetadata", "GetToken", "GetMetadata"},
```

Comment:
Can we verify the last `GetMetadata` call actually uses the `secondToken`, not the `firstToken`.

Reply:
Updated test to validate the expected token vs the one used.